src/node.rs

```diff
@@ -0,0 +1,72 @@
+use num_traits::Float;
+use num_traits::FromPrimitive;
+use num_traits::Zero;
+use std::ops::AddAssign;
+use std::ops::DivAssign;
+use std::ops::MulAssign;
+use std::ops::SubAssign;
+use crate::rect::Rect;
+use crate::sphere::Sphere;
+
+
+pub enum Data<T, const dimension: usize> {
```

Comment:
I'm afraid my response to your question on Slack was misleading.

The dimension doesn't change after training, but in training, users may want to try different dimensions, either because they are dealing with different kinds of data (e.g., daily statistics sampled every hour (24 dimensions) or weekly statistics sampled every day (7 dimensions)), or because they are looking at the same data at different scale through downsampling. Therefore, we cannot fix the dimension at compile-time. However in run-time, it is okay to fix the dimension during training, and later reject any query with a different dimension.

Reply:
Thank you for the clarification. I was afraid of dealing with `dimension` related error checks (literally on every function SRTree offers), so I had to do this: `struct Rect<T, const dimension: usize>`. Ideally, all I need is to create a `const usize` at run-time, which seems impossible in Rust (after reading some SO questions). It seems the error checks are okay as the run-time setting of `dimension` gives scalability. Let me know your thoughts on this:
```
pub struct Rect<T> {
    low: Vec<T>,
    high: Vec<T>,
}
```

Reply:
Looks okay to me.